src/domain/ledger/fee-reimbursement.ts

```diff
@@ -1,12 +1,30 @@
-import { toSats } from "@domain/bitcoin"
+import { WalletCurrency } from "@domain/shared"
 
 import { FeeDifferenceError } from "./errors"
 
-export const FeeReimbursement = (prepaidFee: Satoshis): FeeReimbursement => {
-  const getReimbursement = (actualFee: Satoshis): Satoshis | FeeDifferenceError => {
-    const feeDifference = toSats(prepaidFee - actualFee)
-    if (feeDifference < 0) return new FeeDifferenceError()
-    return feeDifference
+export const FeeReimbursement = ({
+  prepaidFeeAmount,
+  priceRatio,
+}: {
+  prepaidFeeAmount: {
+    btc: BtcPaymentAmount
+    usd: UsdPaymentAmount
+  }
+  priceRatio: PriceRatio
```

Comment:
IIsn't passing in the price Ratio redundant? We're already passing in both `btc` and `usd` so we can construct the price ratio in the function right?

Reply:
I had it like this before, but a ratio drawn from the much smaller fee amounts was less sensitive than one derived from the full payment amounts and I was getting some errors at a point based on this. I added the explicit `priceRatio` to calculate the ratio from the full amounts instead outside of the method and then pass that in alongside the fee amounts.